internal/wclayer/cim/LayerWriter.go

```diff
@@ -231,59 +204,33 @@ func NewCimLayerWriter(ctx context.Context, path string, parentLayerPaths []stri
 		}
 	}()
 	span.AddAttributes(
-		trace.StringAttribute("path", path),
+		trace.StringAttribute("path", layerPath),
+		trace.StringAttribute("cimPath", cimPath),
+		trace.StringAttribute("parentLayerPaths", strings.Join(parentLayerCimPaths, ", ")),
 		trace.StringAttribute("parentLayerPaths", strings.Join(parentLayerPaths, ", ")))
 
 	parentCim := ""
-	cimDirPath := GetCimDirFromLayer(path)
-	if _, err = os.Stat(cimDirPath); os.IsNotExist(err) {
-		// create cim directory
-		if err = os.Mkdir(cimDirPath, 0755); err != nil {
-			return nil, fmt.Errorf("failed while creating cim layers directory: %w", err)
-		}
-	} else if err != nil {
-		return nil, fmt.Errorf("unable to access cim layers directory: %w", err)
-
-	}
-
 	if len(parentLayerPaths) > 0 {
-		parentCim = GetCimNameFromLayer(parentLayerPaths[0])
+		// We only need to provide parent CIM name, it is assumed that both parent CIM
+		// and newly created CIM are present in the same directory.
```

Comment:
Can we add a check to ensure that `filepath.Dir(cimPath) == filepath.Dir(parentLayerCimPaths[0])`?